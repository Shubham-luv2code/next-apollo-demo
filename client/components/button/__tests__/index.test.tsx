import React from "react";
import Button from "../index";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

describe("Button", () => {
  const handleClickSpy = jest.fn();
  it("should match snapshot", () => {
    const component = renderer.create(
      <Button text={"Load More"} handleClick={handleClickSpy} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be disabled", () => {
    const ref = {
      current: {
        scrollIntoView: jest.fn(),
      },
    };
    const { getByText } = render(
      <Button text={"Load More"} handleClick={handleClickSpy} hasMore={false} />
    );
    const button = getByText("Load More");
    expect(button.getAttribute("disabled")).toBeTruthy();
    expect(handleClickSpy).not.toBeCalled();
  });
});
