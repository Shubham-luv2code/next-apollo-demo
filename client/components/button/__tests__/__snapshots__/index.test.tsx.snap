// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="btn"
  disabled={true}
  onClick={[MockFunction]}
>
  Load More
</button>
`;
